src/main/java/vendored/org/apache/dolphinscheduler/common/utils/process/ProcessImplForWin32.java

```diff
@@ -983,6 +932,92 @@ private static void waitForTimeoutInterruptibly(WinNT.HANDLE handle, long timeou
      * Begin Amazon addition.
      */
 
+    /**
+     * Preparation for running process as another user. Populates the given extraInfo if applicable.
+     * @return environment block for CreateProcess* APIs
+     */
+    private String setupRunAsAnotherUser(String username, String password, Map<String, String> envMap,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 942-957 into a separate method.